restapi/stories.go

```diff
@@ -346,29 +364,38 @@ func (api api) DeleteImageInS3(name string) (string, error) {
 	s3Client := s3.New(newSession)
 	bucket := aws.String(os.Getenv("BUCKET_NAME"))
 
-	_, err = s3Client.DeleteObject(&s3.DeleteObjectInput{ // To Do: Only friendly URL being deleted on Backblaze, not the image
-		Bucket: bucket,
-		Key:    aws.String(name),
+	_, err = s3Client.DeleteObject(&s3.DeleteObjectInput{
+		Bucket:    bucket,
+		Key:       aws.String(name),
+		VersionId: aws.String(versionId),
 	})
 	if err != nil {
 		return "", fmt.Errorf("Could not delete image in S3")
 	}
 	return "Image Successfully Deleted", nil
 }
 
-
 func convertImageURLToKeyName(originalURL string) (string, error) {
-	re := regexp.MustCompile(s3KeyNameRegex)
-	match := re.FindAllStringSubmatch(originalURL, 2)
+	match := s3KeyNameRegex.FindAllStringSubmatch(originalURL, 2)
+	if len(match) == 0 {
+		match = s3KeyNameRegexAlternate.FindAllStringSubmatch(originalURL, 2)
+		if len(match) == 0 {
+			return "", fmt.Errorf("Invalid Image Link - Image Key")
+		}
+	}
+	return match[0][1], nil
+}
+
+func convertImageURLToVersionId(originalURL string) (string, error) {
+	match := s3VersionIdRegex.FindAllStringSubmatch(originalURL, 2)
 	if len(match) == 0 {
-		return "", fmt.Errorf("Invalid Image Link")
+		return "", nil
```

Comment:
whats the point of returning an error in this function if its always nil. I think in this case we should return an `fmt.Errorf`